postreise/plot/plot_carbon_bar.py

```diff
@@ -1,51 +1,73 @@
+from powersimdata.scenario.scenario import Scenario
 
+from postreise.analyze.generation.emissions import (
+    generate_emissions_stats,
+    summarize_emissions_by_bus,
+)
 
-def plot_carbon_bar(carbon_by_bus_1, carbon_by_bus_2, scenario_name_1, scenario_name_2):
-    """Makes bar chart of carbon emissions by fuel type for two scenarios.
 
-    :param dict carbon_by_bus_1: emission by type and bus for first scenario
-        as returned by :func:`postreise.analyze.carbon.summarize_carbon_by_bus`.
-    :param dict carbon_by_bus_2: emission by type and bus, for second scenario
-        as returned by :func:`postreise.analyze.carbon.summarize_carbon_by_bus`.
-    :param str scenario_name_1: name of first scenario, for labeling plot.
-    :param str scenario_name_2: name of second scenario, for labeling plot.
+def plot_carbon_bar(*args, labels=None):
+    """Make bar chart of carbon emissions by fuel type for n scenarios.
+
+    :param powersimdata.scenario.scenario.Scenario args: scenario instances.
+    :param list/tuple/set labels: labels on plot. Default to scenario id.
+    :raises ValueError:
+        if ``args`` are not scenario instances.
+        if ``labels`` has a different length than the number of passed scenarios.
+    :raises TypeError: if ``labels`` is not an iterable.
     """
-    plt.rcParams.update({"font.size": 22})
-    objects = (scenario_name_1, scenario_name_2)
-    y_pos = np.arange(len(objects))
-
-    carbon_val = [
-        sum(carbon_by_bus_1["coal"].values()),
-        sum(carbon_by_bus_2["coal"].values()),
-    ]
-    _ = plt.figure(figsize=(7, 2))
-    plt.barh(y_pos, carbon_val, align="center", alpha=0.25, color=["black"])
-    plt.yticks(y_pos, objects)
-    plt.xlabel("Tons")
-    plt.title("Coal: CO$_2$  Emissions", y=1.04)
-    plt.show()
+    if not all([isinstance(a, Scenario) for a in args]):
+        raise ValueError("all inputs must be Scenario objects")
+    if labels is not None and not isinstance(labels, (list, tuple, set)):
+        raise TypeError("labels must be a list/tuple/set")
+    if labels is not None and len(args) != len(labels):
+        raise ValueError("labels must have same length as number of scenarios")
+
+    labels = tuple([s.info["id"] for s in args]) if labels is None else tuple(labels)
 
-    ng_val = [sum(carbon_by_bus_1["ng"].values()), sum(carbon_by_bus_2["ng"].values())]
+    carbon_val = {"coal": [], "ng": []}
+    for i, s in enumerate(args):
+        grid = s.state.get_grid()
+        carbon_by_bus = summarize_emissions_by_bus(generate_emissions_stats(s), grid)
+        carbon_val["coal"].append(sum(carbon_by_bus["coal"].values()))
+        carbon_val["ng"].append(sum(carbon_by_bus["ng"].values()))
 
-    _ = plt.figure(figsize=(7, 2))
-    plt.barh(y_pos, ng_val, align="center", alpha=0.25, color=["purple"])
-    plt.yticks(y_pos, objects)
-    plt.xlabel("Tons")
-    plt.title("Natural Gas: CO$_2$ Emissions", y=1.04)
+    plt.rcParams.update({"font.size": 15})
```

Comment:
This is a little dangerous. We're effectively setting a global variable, which may have downstream effects on other plot functions called in the same python session.

Reply:
It has been removed. I added a `labels_size` optional parameter that is passed to the `fontsize` parameter of `plt.yticks`. The font size of the x-title and title are hardcoded